Compiler warning due to deferred memberof delay

**Issue Description**
We got several warning related to the way the deferred memberof delay is written:
/home/progier/sb/r67020/tst/source/389-ds-base/ldap/servers/slapd/back-ldbm/ldbm_add.c:1461:21: warning: 'delay' may be used uninitialized [-Wmaybe-uninitialized]
We should rewrite that code to avoid that


A clear and concise description of what the bug is.

**Package Version and Platform:**
 - Platform: Fedora
 - Package and version: main

Add any other context about the problem here.

--- a/ldap/servers/slapd/back-ldbm/ldbm_add.c
+++ b/ldap/servers/slapd/back-ldbm/ldbm_add.c
@@ -1450,16 +1450,15 @@ common_return:
         }
         if (!result_sent) {
             int deferred;
-            PRIntervalTime delay;
 
             if (!is_internal) {
                 slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
                 if (deferred) {
-                    delay = PR_MillisecondsToInterval(100);
-                }
-                while (deferred) {
-                    DS_Sleep(delay);
-                    slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
+                    PRIntervalTime delay = PR_MillisecondsToInterval(100);
+                    while (deferred) {
+                        DS_Sleep(delay);
+                        slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
+                    }
                 }
             }
             slapi_send_ldap_result(pb, ldap_result_code, ldap_result_matcheddn, ldap_result_message, 0, NULL);

--- a/ldap/servers/slapd/back-ldbm/ldbm_delete.c
+++ b/ldap/servers/slapd/back-ldbm/ldbm_delete.c
@@ -1531,16 +1531,15 @@ diskfull_return:
         }
         if (!result_sent) {
             int deferred;
-            PRIntervalTime delay;
 
             if (!is_internal) {
                 slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
                 if (deferred) {
-                    delay = PR_MillisecondsToInterval(100);
-                }
-                while (deferred) {
-                    DS_Sleep(delay);
-                    slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
+                    PRIntervalTime delay = PR_MillisecondsToInterval(100);
+                    while (deferred) {
+                        DS_Sleep(delay);
+                        slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
+                    }
                 }
             }
             slapi_send_ldap_result(pb, ldap_result_code, NULL, ldap_result_message, 0, NULL);

--- a/ldap/servers/slapd/back-ldbm/ldbm_modify.c
+++ b/ldap/servers/slapd/back-ldbm/ldbm_modify.c
@@ -1168,16 +1168,15 @@ common_return:
         if (!result_sent) {
             /* result is already sent in find_entry. */
             int deferred;
-            PRIntervalTime delay;
 
             if (!is_internal) {
                 slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
                 if (deferred) {
-                    delay = PR_MillisecondsToInterval(100);
-                }
-                while (deferred) {
-                    DS_Sleep(delay);
-                    slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
+                    PRIntervalTime delay = PR_MillisecondsToInterval(100);
+                    while (deferred) {
+                        DS_Sleep(delay);
+                        slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
+                    }
                 }
             }
             slapi_send_ldap_result(pb, ldap_result_code, NULL, ldap_result_message, 0, NULL);

--- a/ldap/servers/slapd/back-ldbm/ldbm_modrdn.c
+++ b/ldap/servers/slapd/back-ldbm/ldbm_modrdn.c
@@ -1510,17 +1510,15 @@ common_return:
         }
         if (!result_sent) {
             int deferred;
-            PRIntervalTime delay;
 
             if (!is_internal) {
-                /* for direct operation, wait for members update */
                 slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
                 if (deferred) {
-                    delay = PR_MillisecondsToInterval(100);
-                }
-                while (deferred) {
-                    DS_Sleep(delay);
-                    slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
+                    PRIntervalTime delay = PR_MillisecondsToInterval(100);
+                    while (deferred) {
+                        DS_Sleep(delay);
+                        slapi_pblock_get(pb, SLAPI_DEFERRED_MEMBEROF, &deferred);
+                    }
                 }
             }
             slapi_send_ldap_result(pb, ldap_result_code, ldap_result_matcheddn,

